### ISSUE TITLE
[Basic] Research for different solutions for Text to Video and compare the advantages and disadvantages

### ISSUE THREAD

**Issue by yuanxion** [yuanxion/Text2Video-Zero#3]

**Comment by yuanxion** [comment]
Read papers to get more (innovation) ideas.
Share learning of Diffusion model/ControlNet.

Idea: Make the video generation more smoothing.

**Comment by cold-blue** [comment]
Hi, all. I wonder whether it is possible to compute a 3D consistency metric between different frames? Maybe this kind of metric needs us to propose in some special format.

**Comment by aixiaodewugege** [comment]
Maybe one way is trying to use these frames and nerf to generate a 3D scene. And analysis whether the 3D scene is good?

**Comment by yuanxion** [comment]
- 3D Metrics

3D reconstruction tasks are typically evaluated using reference-based metrics like **Chamfer Distance** (https://zhuanlan.zhihu.com/p/629086843):

<img width="578" alt="image" src="https://github.com/yuanxion/Text2Video-Zero/assets/96522341/b0d10502-40d9-4f09-ab04-55678e096f4d">

From DreamFusion, they evaluate the **CLIP R-Precision**: https://arxiv.org/pdf/2209.14988.pdf.

<img width="560" alt="image" src="https://github.com/yuanxion/Text2Video-Zero/assets/96522341/a856369f-4bd4-4d74-8ef6-447c99a4a4b4">

**Comment by cold-blue** [comment]
I have a consideration: maybe we shouldn't choose Zero to be the base model for our second stage. The advantage of Zero is that it doesn't need any training. If we break this constraint to enhance motion coherence by finetuning it in some way, it may be meaningless in terms of innovation because there have been some models solving the motion coherence problem by finetuning a text-to-image model with unlabeled video datasets. If we indeed want to solve the coherence problem by 3D latent constraints, maybe we should choose some models which are designed to be finetuned to be our base model or comparison ones. However, for the accelerating task in our first stage, Zero is a good choice.

**Comment by cold-blue** [comment]
https://arxiv.org/pdf/2303.11328.pdf
![MicrosoftTeams-image (4)](https://github.com/yuanxion/Text2Video-Zero/assets/21053269/c62cecbd-ca83-40f6-9585-49fef56a952f)

**Comment by cold-blue** [comment]
https://arxiv.org/abs/2303.14184
![MicrosoftTeams-image (5)](https://github.com/yuanxion/Text2Video-Zero/assets/21053269/7bc4904c-e5af-4c0d-b4da-1db01cffa36f)

**Comment by aixiaodewugege** [comment]
We can also generate 3D motion, if we only focus on human video.
![image](https://github.com/yuanxion/Text2Video-Zero/assets/47085237/8cc4b9df-f759-4e8d-91da-05a30a387cf2)

paper :  Generating Diverse and Natural 3D Human Motions from Text

**Comment by yuanxion** [comment]
> I have a consideration: maybe we shouldn't choose Zero to be the base model for our second stage. The advantage of Zero is that it doesn't need any training. If we break this constraint to enhance motion coherence by finetuning it in some way, it may be meaningless in terms of innovation because there have been some models solving the motion coherence problem by finetuning a text-to-image model with unlabeled video datasets. If we indeed want to solve the coherence problem by 3D latent constraints, maybe we should choose some models which are designed to be finetuned to be our base model or comparison ones. However, for the accelerating task in our first stage, Zero is a good choice.

Yes, agree. We have powerful GPUs with Server now, so it is ok to change and train whichever models.

**Comment by cold-blue** [comment]
https://huggingface.co/blog/text-to-video
In this web page, there are several model demos including Zero for us to try freely.
<img width="596" alt="Screenshot 2023-06-08 101421" src="https://github.com/yuanxion/Text2Video-Zero/assets/21053269/84710310-42d1-458e-8590-020e6816dd55">

**Comment by cold-blue** [comment]
I have tried several T2V demos on huggingface and this one seems the best up to now.
https://huggingface.co/spaces/NeuralInternet/Text-to-Video_Playground
<img width="1053" alt="Screenshot 2023-06-08 101506" src="https://github.com/yuanxion/Text2Video-Zero/assets/21053269/873b44ad-a564-4740-a991-8d26c1eec4d0">
